### PR TITLE
DOC: Fix type of axis in nanfunctions

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Hi, I'm contributing to numpy for the first time, so please let me know if you need more info. This is a simple doc-fix for some functions within `nanfunctions.py` regarding the type of the `axis` argument.

### CODE REVIEW

**Comment by charris** [comment]
Close and reopen to restart tests.

**Comment by eric-wieser** [comment]
Matches the docstring for `nanpercentile`, so looks good to me

File: numpy/lib/nanfunctions.py [hunk 1]
```diff
@@ -458,7 +458,7 @@ def nanargmax(a, axis=None):
     ----------
     a : array_like
         Input data.
-    axis : int, optional
+    axis : {int, tuple of int, None}, optional
```
Comment (eric-wieser) [inline numpy/lib/nanfunctions.py]: This is incorrect - it only accepts integers

File: numpy/lib/nanfunctions.py [hunk 2]
```diff
@@ -726,7 +726,7 @@ def nancumprod(a, axis=None, dtype=None, out=None):
     ----------
     a : array_like
         Input array.
-    axis : int, optional
+    axis : {int, tuple of int, None}, optional
```
Comment (eric-wieser) [inline numpy/lib/nanfunctions.py]: Incorrect

**Comment by eric-wieser** [comment]
Maybe you should add some tests to prove each of these claims

**Review by eric-wieser — DISMISSED** [review]
As commented above, some of this documentation isn't correct.

Maybe you should add some tests to prove whether each of these takes tuple arguments.

**Comment by Tobias-Fischer** [comment]
Apologies for that - the type of the `axis` argument is now fixed in the last commit (now matching those of the "non-nan" equivalent methods). 
Regarding the addition of tests: this is beyond the scope of this PR (at least for me, if someone else wants to contribute regarding tests, great). This PR is only aimed at fixing some docs, and does not aim to increase test coverage.

File: numpy/lib/nanfunctions.py [hunk 3]
```diff
@@ -306,7 +306,7 @@ def nanmax(a, axis=None, out=None, keepdims=np._NoValue):
     a : array_like
         Array containing numbers whose maximum is desired. If `a` is not an
         array, a conversion is attempted.
-    axis : int, optional
+    axis : {int, tuple of int, None}, optional
         Axis along which the maximum is computed. The default is to compute
```
Comment (eric-wieser) [inline numpy/lib/nanfunctions.py]: These should all be updated to say `Axis or axes`, which would match the docs for `sum` and others

**Comment by eric-wieser** [comment]
Thanks @Tobias-Fischer

**Pull request merged**